### PR TITLE
aggregation logic and other bug fixes

### PR DESCRIPTION
**Pull request by nithinj**
* qsub should make use of microsecond seed for randomness
* server should not acknowledge sched after async run job
* Source server should purge job if it is getting job already exists error while move and run.
* job count aggregation
* resources assigned aggregation

### CODE REVIEW

File: src/lib/Libutil/misc_utils.c [hunk 1]
```diff
@@ -2210,3 +2210,18 @@ get_current_servers()
 {
 	return pbs_conf.pbs_current_servers;
 }
+
+int
+rand_num()
```
Comment (agrawalravi90) [inline src/lib/Libutil/misc_utils.c:2215]: Please add a function header

File: src/lib/Libifl/int_status.c [hunk 2]
```diff
@@ -125,6 +203,204 @@ PBSD_status(int c, int function, char *objid, struct attrl *attrib, char *extend
...
+		else if (pc && *pc != '\0')
+			*type = STRING;
+	} else {
+		if ((*val_long = strtol(val, &pc, 10)))
+			*type = LONG;
+		else if (pc && *pc != '\0')
+			*type = STRING;
+	}
+	if (*type == STRING) {
+		if ((*pc == 'k' || *pc == 'm' || *pc == 'g' || *pc == 't' || *pc == 'p') && pc[1] == 'b') {
```
Comment (agrawalravi90) [inline src/lib/Libifl/int_status.c:273]: you need to do strcasecmp(pc, "kb"), strcasecmp(pc, "mb"), etc., *pc == 'k'  will also satisfy a string value like "kardashian" :)

File: src/lib/Libifl/int_status.c [hunk 3]
```diff
@@ -125,6 +203,204 @@ PBSD_status(int c, int function, char *objid, struct attrl *attrib, char *extend
...
+			*type = SIZE;
+			*val_long = strtol(val, &pc, 10);
```
Comment (agrawalravi90) [inline src/lib/Libifl/int_status.c:275]: you already did this on line 267, val_long should already be set.

File: src/include/batch_request.h [hunk 4]
```diff
@@ -141,6 +141,7 @@ struct rq_move {
 	char rq_jid[PBS_MAXSVRJOBID + 1];
 	char rq_destin[(PBS_MAXSVRRESVID > PBS_MAXDEST ? PBS_MAXSVRRESVID : PBS_MAXDEST) + 1];
 	char *run_exec_vnode;
+	int orig_rq_type;
```
[the remark this answers is not in the document]
Comment (nithinj) [reply src/include/batch_request.h:144]: We use the same runjob request coming from sched as a movejob request. So this field is useful in checking whether to send a reply back to sched.

File: src/lib/Libifl/int_status.c [hunk 5]
```diff
@@ -46,13 +46,92 @@
...
+			d = &count[HELD_STATE];
+		else if (strcmp(s, statename[WAIT_STATE]) == 0)
+			d = &count[WAIT_STATE];
+		else if (strcmp(s, statename[TRANSIT_STATE]) == 0)
+			d = &count[TRANSIT_STATE];
+		else if (strcmp(s, statename[EXITING_STATE]) == 0)
+			d = &count[EXITING_STATE];
+		else if (strcmp(s, statename[BEGUN_STATE]) == 0)
+			d = &count[BEGUN_STATE];
+		c++;
```
[the remark this answers is not in the document]
Comment (nithinj) [reply src/lib/Libifl/int_status.c:109]: You meant to say back to colon. I will do it. In out case we were not using this string anyway. So it was fine.

File: src/lib/Libifl/int_status.c (hunk 2 above)
Comment (nithinj) [reply src/lib/Libifl/int_status.c:273]: We wont hit "kardashian" as we also check for pc[1] == 'b'
But I should make sure it ends with that.

**Pull request merged**